rskj-core/src/test/java/co/rsk/net/NodeMessageHandlerTest.java

```diff
@@ -205,6 +211,105 @@ void postBlockMessageFromBannedMiner() {
+                config,
+                sbp,
+                syncProcessor,
+                null,
+                channelManager,
+                null,
+                scoring,
+                statusResolver,
+                Mockito.mock(Thread.class),
+                messageCounter
+        ));
+
+        nodeMessageHandler.postMessage(sender, snapStateChunkRequestMessage, null);
+        nodeMessageHandler.postMessage(sender, snapStateChunkResponseMessage, null);
+        nodeMessageHandler.postMessage(sender, snapStatusRequestMessage, null);
+        nodeMessageHandler.postMessage(sender, snapStatusResponseMessage, null);
+        nodeMessageHandler.postMessage(sender, snapBlocksRequestMessage, null);
+        nodeMessageHandler.postMessage(sender, snapBlocksResponseMessage, null);
+
+        Assertions.assertEquals(6, nodeMessageHandler.getMessageQueueSize());
+
+        ArgumentCaptor<Message> snapMessagesCaptor = ArgumentCaptor.forClass(Message.class);
+
+        nodeMessageHandler.start();
+        // Snap responses scores = 300
+        // Snap requests scores = 100
+
+        nodeMessageHandler.run();
+        Mockito.verify(nodeMessageHandler, atLeastOnce()).processMessage(any(Peer.class), snapMessagesCaptor.capture());
+        Assertions.assertEquals(MessageType.SNAP_STATE_CHUNK_RESPONSE_MESSAGE, snapMessagesCaptor.getValue().getMessageType());
+
+        nodeMessageHandler.run();
+        Mockito.verify(nodeMessageHandler, atLeastOnce()).processMessage(any(Peer.class), snapMessagesCaptor.capture());
+        Assertions.assertEquals(MessageType.SNAP_STATUS_RESPONSE_MESSAGE, snapMessagesCaptor.getValue().getMessageType());
+
+        nodeMessageHandler.run();
+        Mockito.verify(nodeMessageHandler, atLeastOnce()).processMessage(any(Peer.class), snapMessagesCaptor.capture());
+        Assertions.assertEquals(MessageType.SNAP_BLOCKS_RESPONSE_MESSAGE, snapMessagesCaptor.getValue().getMessageType());
+
+        nodeMessageHandler.run();
+        Mockito.verify(nodeMessageHandler, atLeastOnce()).processMessage(any(Peer.class), snapMessagesCaptor.capture());
+        Assertions.assertEquals(MessageType.SNAP_STATE_CHUNK_REQUEST_MESSAGE, snapMessagesCaptor.getValue().getMessageType());
+
+        nodeMessageHandler.run();
+        Mockito.verify(nodeMessageHandler, atLeastOnce()).processMessage(any(Peer.class), snapMessagesCaptor.capture());
+        Assertions.assertEquals(MessageType.SNAP_BLOCKS_REQUEST_MESSAGE, snapMessagesCaptor.getValue().getMessageType());
+
+        nodeMessageHandler.run();
+        Mockito.verify(nodeMessageHandler, atLeastOnce()).processMessage(any(Peer.class), snapMessagesCaptor.capture());
+        Assertions.assertEquals(MessageType.SNAP_STATUS_REQUEST_MESSAGE, snapMessagesCaptor.getValue().getMessageType());
+
+        nodeMessageHandler.stop();
+
+        Assertions.assertEquals(0, nodeMessageHandler.getMessageQueueSize());
```

Comment:
if any of these fail, how do you know which one it was that did so?
I'd argue that all of these should be in their own test functions to ensure that they cannot affect each other. You might say that you know that they don't. But 1. you don't know that they won't, and 2. you should not use your knowledge of the code to create the tests, as that is just writing the same thing twice in effect and not actually testing anything.
But then, I see this done so much across this codebase.

Reply:
The purpose of this test is to verify that when we post various snap messages, they are correctly added in the right order based on their priority, which is determined by the Stats.score() method defined. In this test, we're focused on ensuring that the priority and score assignments are accurate. We're not concerned with whether the messages fail or not; our goal is solely to validate the correct setting of priority and score. I apologize if this wasn't clear from the test's name. I'll rename it to something more descriptive to avoid any confusion

Reply:
renamed the test to `checkSnapMessagesOrderAndPriority()`